### PR TITLE
Feature/lessons overview

### PR DESCRIPTION
**Pull request by MPepito**
### PR description 

The goal of this PR is to introduces the homeScreen that displays the user's lessons based on the user's profile, provides role-based navigation for Tutors and Students to find a Lesson and provide an Icon in the top bar to navigate to the screen where the user can view its profil information. 

### Summary:

- homeScreen Composable: Displays a list of the user's lessons, a message when no lessons are available, or an error screen when no profile is assigned to the current user and has an Icon on the top bar that allow the user to navigate to the information of its profile.
- Role-based Navigation: The bottom navigation bar dynamically updates based on whether the user is a Tutor or Student.
- Lesson Display: Lessons are presented in a card format, with details like the title, time slot, user's first name, and lesson price but design remain to be done by @AlexisAllmnd. Currently the two lessons displayed are hard coded such that one can view the design of a lesson. Once @Philipp0206 is done with #45 it can be link by uncommenting the line "val lessons" in Overview.kt
- Navigation Actions Update: Added support for separate top-level destinations for Tutors and Students in the navigation logic.
- MainActivity Update: Handle different navigation and routing based on the user's role (Student or Tutor) using the NavigationActions object.

Once the ProfilInfoScreen is available we can test the navigation to the information of the profil when clicking on the profil Icon.

### How to test

To test this PR, one can run the MainActivity and once the profil is created, one can check that the screen (Screen.HOME) corresponds to the screenshots given below and when clicking on the "My courses" and "Find a Tutor" the user will nagivate to two different routes that only contains at the moment the home screen (Screen.HOME). The app will crash when we click on the Profil Icon on the Top right as it is already set to navigate to Screen.PROFILE but it requires the composable #43 from @alexs-rufer that is ready in a few hours. 

### Screenshots
Screen.HOME:
WhatsApp Image 2024-10-16 at 11:36:07

### CODE REVIEW

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      // Lesson Title at the top
+      Text(text = lesson.title)
+
+      Spacer(modifier = Modifier.height(4.dp))
+
+      // Row to display Lesson.timeSlot, currentProfileFirstName, and Lesson.price on the same line
+      Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+        Text(text = currentProfile.academicLevel.toString())
+        Text(text = "\$${lesson.price}")
+        Text(text = lesson.timeSlot)
```
Comment (AlexisAllmnd) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:157]: On the comment you say profile name but on the code you putted academic Level.

**Review by AlexisAllmnd — COMMENTED** [review]
Overall your changes are good and the code is clear and relevant. Just a little detail about a confusion between the comments and your code.

**Review by AlexisAllmnd — APPROVED** [review]
Great work on implementing the lesson overview feature! The code is clean and follows good practices, especially with the separation of concerns for retrieving lessons based on the user's role. I appreciate the effort you've put into refactoring the repository methods for better clarity and reuse.

I'll take it from here and work on improving the UI to enhance the user experience.

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    Role.UNKNOWN -> Toast.makeText(context, "Unknown Profil", Toast.LENGTH_SHORT).show()
+    null -> Toast.makeText(context, "null Profil", Toast.LENGTH_SHORT).show()
+  }
+
+  // Determine which bottom navigation items to show based on the user's role
+  val LIST_TOP_LEVEL_DESTINATIONS =
+      when (currentProfile?.role) {
+        Role.TUTOR -> LIST_TOP_LEVEL_DESTINATIONS_TUTOR
+        Role.STUDENT -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+        Role.UNKNOWN -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:63]: maybe in that "UNKNOWN" case (that should never happen) we could use an empty list and show a toast, but this is not mandatory for the moment !

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+  Scaffold(
+      topBar = {
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(16.dp),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically) {
+              IconButton(
+                  onClick = {
+                    Toast.makeText(context, "Profile icon clicked", Toast.LENGTH_SHORT).show()
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:75]: now that the navigation is implemented, we might want to remove that toast :)

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+              if (currentProfile != null) {
+                LessonItem(
+                    currentProfile,
+                    lesson = lessons[index],
+                    onClick = {
+                      // Logic for selecting a lesson and navigating to the edit screen
+                      lessonViewModel.selectLesson(lessons[index])
+                      // TODO: nagivate to edit lesson screen
+                      Toast.makeText(context, "Navigate to EDIT_LESSON screen", Toast.LENGTH_LONG)
+                          .show()
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:104]: same comment here,  we might want to remove the toast. as the edit lesson is note implemented yet, we can keep it for now!

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                      // navigationActions.navigateTo(Route.EDIT_LESSON)
+                    },
+                    modifier = Modifier.padding(vertical = 8.dp))
+              } else {
+                Toast.makeText(
+                        context,
+                        "No profil is currently assigned to the current user",
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:111]: typo here "profile" not "profil"

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                        Toast.LENGTH_LONG)
+                    .show()
+                Log.e("homeScreen", "No profil is currently assigned to the current user")
```
Comment (alexs-rufer) [inline app/src/main/java/com/github/se/project/ui/overview/Overview.kt:114]: same

**Review by alexs-rufer — APPROVED** [review]

File: app/src/main/java/com/github/se/project/ui/overview/Overview.kt (hunk 3 above)
Comment (MPepito) [reply app/src/main/java/com/github/se/project/ui/overview/Overview.kt:75]: Good point

**Review by alexs-rufer — APPROVED** [review]
Great job, we can now merge this branch to main :)

**Comment by MPepito** [comment]
feature/add lesson, refactor/profil and feature/profil screen were all merged on that branch before mergin with main

**Pull request merged**